firebase-common/src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -324,23 +321,14 @@ public static FirebaseApp initializeApp(
   }
 
   /**
-   * Specify which services should access local emulators for this FirebaseApp instance.
-   *
-   * <p>For example, if the {@link EmulatorSettings} contain {@link
-   * com.google.firebase.emulators.EmulatedServiceSettings} for {@link FirebaseDatabase#EMULATOR},
-   * then calls to Cloud Firestore will communicate with the emulator rather than production.
-   *
-   * <p>TODO(samstern): Un-hide this once Firestore, Database, and Functions are implemented
+   * This should <b>only</b> ever be called from inside FirebaseFoo.useEmulator().
```

Comment:
If it should now only be called by `useEmulator()` methods, why does FirebaseApp need to host this functionality as opposed to each sdk handling it individually?

Reply:
If we want to have a central place for this config but it does not require a public API, then the recommended approach is to use Components as opposed to adding functionality to `FirebaseApp`

Reply:
I guess there's no need to centralize it anymore.  We may want it one day, but that's for then I guess.

Reply:
Actually it's pretty hard to make `FirebaseDatabase.getInstance(app, url)` work without this centralization ... so I'll do a component.

Reply:
hm actually, rtdb has [FirebaseDatabaseComponent](https://github.com/firebase/firebase-android-sdk/blob/master/firebase-database/src/main/java/com/google/firebase/database/FirebaseDatabaseComponent.java), could it be used to host the emulator settings as opposed to having a component in firebase-common?

Reply:
Done.